File: engine/engine-model/src/main/scala/net/server/channel/packet/ring/InviteToWeddingPacket.scala
package net.server.channel.packet.ring

class InviteToWeddingPacket(private var _mode: Byte, private var _name: String, private var _marriageId: Int, private var _slot: Byte) extends BaseRingPacket(_mode = _mode) {
  def name: String = _name

  def marriageId: Int = _marriageId

  def slot: Byte = _slot
}

File: engine/engine-model/src/main/java/net/server/channel/packet/ring/InviteToWeddingPacket.java
package net.server.channel.packet.ring;

public class InviteToWeddingPacket extends BaseRingPacket {
   private final String name;

   private final Integer marriageId;

   private final Byte slot;

   public InviteToWeddingPacket(Byte mode, String name, Integer marriageId, Byte slot) {
      super(mode);
      this.name = name;
      this.marriageId = marriageId;
      this.slot = slot;
   }

   public String name() {
      return name;
   }

   public Integer marriageId() {
      return marriageId;
   }

   public Byte slot() {
      return slot;
   }
}
